entrypoint.py

```diff
@@ -42,6 +42,11 @@ def main() -> None:
             python_names = KNOWN_PYTHONS[KNOWN_PYTHONS.index(min_python) : (KNOWN_PYTHONS.index(max_python)+1)]
         python_flavours = len(python_names)
         for env in other_names:
+            # Check for using correct python version for other_names like py310-devel.
+            match = re.search('py(\d+)', env)
+            if match:
+                py_version = match.groups()[0]
+                default_python = f"{py_version[0]}.{py_version[1:]}"
```

Comment:
Here you override the value of default_python which will remain changed even after the loop runs. This will change the default_python for environments where it should not.